### PR TITLE
Fallback for localIpAddr failure

### PR DESCRIPTION
**Pull request by rvanderwal-newrelic**
When running the JFR daemon in an Ubuntu or Alpine container on an Amazon EC2 instance, the JRE inside the container can't resolve the short hostname it receives to do a reverse DNS lookup. This detects failure of `getLocalHost()` and just puts in a localhost IP address.

You probably don't want to accept this PR, since it has a hard-coded IP address in it. I'd love to see how _you'd_ do the same thing.

### CODE REVIEW

File: jfr-daemon/src/main/java/com/newrelic/jfr/daemon/JFRController.java [hunk 1]
```diff
@@ -142,7 +142,13 @@ private static DaemonConfig buildConfig() {
 
   static JFRUploader buildUploader(DaemonConfig config)
       throws UnknownHostException, MalformedURLException {
-    String localIpAddr = InetAddress.getLocalHost().toString();
+    String localIpAddr;
+    try {
+      localIpAddr = InetAddress.getLocalHost().toString();
+    } catch (Throwable e) {
+      logger.error("Unable to get localhost IP, defaulting to 127.0.0.1.", e);
+      localIpAddr = "127.0.0.1";
```
Comment (jeffalder) [inline jfr-daemon/src/main/java/com/newrelic/jfr/daemon/JFRController.java:150]:
```suggestion
      localIpAddr = InetAddress.getLoopbackAddress();
```

This is what we do in the agent.

Comment (rvanderwal-newrelic) [reply jfr-daemon/src/main/java/com/newrelic/jfr/daemon/JFRController.java]: Thanks, that looks more legitimate.

File: jfr-daemon/src/main/java/com/newrelic/jfr/daemon/JFRController.java [hunk 2]
```diff
@@ -142,7 +142,13 @@ private static DaemonConfig buildConfig() {
 
   static JFRUploader buildUploader(DaemonConfig config)
       throws UnknownHostException, MalformedURLException {
-    String localIpAddr = InetAddress.getLocalHost().toString();
+    String localIpAddr;
+    try {
+      localIpAddr = InetAddress.getLocalHost().toString();
+    } catch (Throwable e) {
+      logger.error("Unable to get localhost IP, defaulting to 127.0.0.1.", e);
+      localIpAddr = InetAddress.getLoopbackAddress();
```
Comment (kittylyst) [inline jfr-daemon/src/main/java/com/newrelic/jfr/daemon/JFRController.java:150]: Shouldn't this be .toString() here?

Comment (rvanderwal-newrelic) [reply jfr-daemon/src/main/java/com/newrelic/jfr/daemon/JFRController.java]: Yeah, fix incoming soon.

**Pull request merged**